api.go

```diff
@@ -511,3 +522,14 @@ func (api *CacophonyAPI) GetSchedule() ([]byte, error) {
 
 	return ioutil.ReadAll(resp.Body)
 }
+
+type notRegisteredError struct{}
+
+func (e *notRegisteredError) Error() string {
+	return "device is not registered"
+}
```

Comment:
This is fine but you could also have created a singleton instance and checked for that in IsNotRegisteredError, like this:

```go
var notRegisteredError = errors.New("device is not registered")

func IsNotRegisteredError(err error) bool {
	return err == notRegisteredError
}
```

This makes returning the error a little nicer too. Just `return notRegisteredError`

Not a biggie though. The nice thing about making it a private error type like you've done, and checking with IsNotRegisteredError, is that the approach can be changed without affecting the callers.